binaryIoTypeC/proto/binaryIoTypeC/v1alpha1/binaryIoTypeC.proto

```diff
@@ -0,0 +1,220 @@
+message FunctionControlSet {
+    oneof type {
+        SetSingle single = 1;
+        SetAll all = 2;
+    }
+}
+
+// Channel diagnostic bits. Multiple bits may be set at the same time.
+enum ChannelDiag {
+    NoDiag = 0;
+    // group supply voltage missing (input values can't be determined, output values can't be set)
+    NoSupplyVoltage = 1;
+    // channel current limiter is limiting current
+    CurrentLimit = 2;
+    // channel or group overload (output is disabled, output can't be set)
+    Overload = 4;
+    // group supply undervoltage
+    SupplyUndervoltage = 8;
+    // group supply overvoltage
+    SupplyOvervoltage = 16;
+}
+
+message GetSingleResponse {
+    // Number of the binary input channel
+    fixed32 channel = 1;
+    // State for the bianry input channel
+    bool state = 2;
+    // channel diagnostic bits
+    fixed32 diag = 3;
+}
+
+message GetAllResponse {
+    // binary coded map of input values. 0 means low, 1 means high, LSB is channel 0.
+    // In case the input value is unknown, its value is reported as 0. The input value is invalid if
+    // the diag NoSupplyVoltage bit is set.
+    fixed32 inputs = 1;
+
+    // diagnostic information for each channel, containing the ChannelDiag bits.
+    // first diag corresponds to channel 0, second to channel 1, etc.
+    repeated uint32 diag = 2;
+}
+
+// FunctionControlGetResponse to pass to Functionblock.FunctionControlResponse.functionSpecificControlGet hook
+message FunctionControlGetResponse {
+    oneof type {
+        GetSingleResponse single = 1;
+        GetAllResponse all = 2;
+    }
+}
+
+message SetSingleResponse {
+    // no parameters
+}
+
+message SetAllResponse {
+    // no parameters
+}
+
+// FunctionControlSetResponse to pass to Functionblock.FunctionControlResponse.functionSpecificControlSet hook
+message FunctionControlSetResponse {
+    oneof type {
+        SetSingleResponse single = 1;
+        SetAllResponse all = 2;
+    }
+}
+
+// ============= StreamControl ==================
+// StreamControlStart to pass to Functionblock.StreamControlStart.functionSpecificStreamControlStart hook
+message StreamControlStart {
+    // mask to define which channels are considered in the stream. 0 means don't send channels value, 1 means sent it
+    // LSB is channel 0
+    fixed32 channelFilterMask = 1;
+}
+
+// Sample is taken whenever the values change with respect to the last reported sample on that stream.
+// An initial sample is sent immediately after stream start.
+message Sample {
+    // Timestamp for that specific channels sample. This is the time the sample was taken.
+    // This timestamp is in microseconds since the start of the device and does not get synchronized with the clients time.
+    fixed64 timestamp = 1;
+    // binary coded map of input values. 0 means low, 1 means high, LSB is channel 0.
+    // In case the input value is unknown, its value is reported as 0.
+    fixed32 values = 2;
```

Comment:
Why is the diag field not sent with the stream?

Reply:
Because it would consume too much space in the stream. Maybe I should add one more bitfield that tells whether the channels value is valid?